core/src/cluster_info.rs

```diff
@@ -2272,95 +2276,114 @@ impl ClusterInfo {
     fn handle_batch_push_messages(
         &self,
         messages: Vec<(Pubkey, Vec<CrdsValue>)>,
+        thread_pool: &ThreadPool,
         recycler: &PacketsRecycler,
         stakes: &HashMap<Pubkey, u64>,
         response_sender: &PacketSender,
     ) {
-        for (from, data) in messages {
-            let response = self.handle_push_message(recycler, &from, data, stakes);
-            if let Some(response) = response {
-                let _ = response_sender.send(response);
-            }
+        if messages.is_empty() {
+            return;
         }
-    }
-
-    fn handle_push_message(
-        &self,
-        recycler: &PacketsRecycler,
-        from: &Pubkey,
-        mut crds_values: Vec<CrdsValue>,
-        stakes: &HashMap<Pubkey, u64>,
-    ) -> Option<Packets> {
-        let self_id = self.id();
-        self.stats.push_message_count.add_relaxed(1);
-        let len = crds_values.len();
-
-        let shred_version = self
-            .lookup_contact_info(from, |ci| ci.shred_version)
-            .unwrap_or(0);
-        Self::filter_by_shred_version(
-            from,
-            &mut crds_values,
-            shred_version,
-            self.my_shred_version(),
-        );
-        let filtered_len = crds_values.len();
+        self.stats
+            .push_message_count
+            .add_relaxed(messages.len() as u64);
+        // Obtain shred versions of the origins.
+        let shred_versions: Vec<_> = {
+            let gossip = self.gossip.read().unwrap();
+            messages
+                .iter()
+                .map(|(from, _)| match gossip.crds.get_contact_info(from) {
+                    None => 0,
```

Comment:
I think it's more idiomatic to use combinators over a match here, i.e.

```
 messages.iter().map(|(from, _)|
                gossip
                    .crds
                    .get_contact_info(from)
                    .map_or(0, |info| info.shred_version)
```

Reply:
I personally find the match syntax more readable here. It also avoids additional call-stacks (`map_or` plus the closure), and the rust formatter adds 2 additional lines.